### PR TITLE
feat: [P4PU-213] Error and Empty state for payment notices route. 

### PR DESCRIPTION
**Pull request by AnisLucidi01**
#### List of Changes

Added empty state and error state component and logic to payment notices page.
#### Motivation and Context
This change is required by design.

#### How Has This Been Tested?
Unit testing, visual testing. 

#### Screenshots (if appropriate):
image
image

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/routes/PaymentNotices.tsx [hunk 1]
```diff
@@ -3,116 +3,90 @@ import Stack from '@mui/material/Stack';
...
+      <QueryLoader
+        // TODO: fallback component of behavior be defined
+        fallback={isError && <PaymentNotice.Error />}
+        queryKey="paymentNotices">
+        <Stack gap={5} component="section">
+          <Typography paddingTop={3} variant="h4" component="h1">
+            {t('app.paymentNotices.title')}
+          </Typography>
+          {data.length > 0 ? (
+            <Stack gap={3}>
```
Comment (stratoivandiluccio) [inline src/routes/PaymentNotices.tsx:28]: What about creating a new component using liner from 28 to 62 so we could rewrite this ternary in a more elegant and concise way?

I mean something like that:
`data.length > 0 ? <Payments.ok data={data} /> : <Payments.empty />`

File: src/components/PaymentNotice/Empty.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        <Stack spacing={2} width={'100%'} textAlign={'center'} alignItems={'center'}>
+          <Typography variant="body1" fontWeight={700}>
+            {t('app.paymentNotice.empty.title')}
+          </Typography>
+          <Typography variant="body1">{t('app.paymentNotice.empty.description')}</Typography>
+          <Button
+            variant="contained"
+            role="button"
+            aria-label={t('app.paymentNotice.empty.button')}
+            sx={{ maxWidth: '60%', marginTop: '24px !important' }}
```
Comment (vito80ba) [inline src/components/PaymentNotice/Empty.tsx:32]: "important"... is it necessary?

File: src/components/PaymentNotice/Error.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      }}>
+      <CardActions>
+        <Stack spacing={2} width={'100%'} textAlign={'center'} alignItems={'center'}>
+          <ErrorOutline color="error" />
+          <Typography variant="body1">{t('app.paymentNotice.error.description')}</Typography>
+          <Button
+            variant="text"
+            aria-label={t('app.paymentNotice.error.button')}
+            role="button"
+            sx={{ maxWidth: '60%', marginTop: '24px !important' }}
```
Comment (vito80ba) [inline src/components/PaymentNotice/Error.tsx:31]: Same as above

Comment (AnisLucidi01) [reply src/components/PaymentNotice/Error.tsx:31]: Unfortunately yes, I haven't been able to understand why.

**Review by vito80ba — APPROVED** [review]

**Pull request merged**